usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateAllOptionalFileName.java

```diff
@@ -0,0 +1,40 @@
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.mobilitydata.gtfsvalidator.usecase.notice.OptionalFileFoundNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsSpecRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.RawFileRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.List;
+
+public class ValidateAllOptionalFileName {
+
+    private final GtfsSpecRepository specRepo;
+    private final RawFileRepository rawFileRepo;
+    private final ValidationResultRepository resultRepo;
+    private final List<String> filenameList;
+
+    public ValidateAllOptionalFileName(final GtfsSpecRepository specRepo, final RawFileRepository rawFileRepo,
```

Comment:
This and `execute()` need Javadocs. It's not clear from the calling class that the `filenameList` you pass in here will be appended to and returned from `execute()`. Also, you need to decide here whether you should really be modifying the list that is passed in, or if you should first create a copy of the provided list and then modify the copy. The way this is currently set up, calling code may not expect that the list that is passed into the constructor will be modified. Either way, the behavior should be documented.

Reply:
Excellent point. We will modify the behavior as follow
- `ValidateAllRequiredFilePresence` will return the list of all required filenames
- `ValidateAllOptionalFilename`will return the list of all validated (that is, present in the schema file) optional filename
Concatenation will happen in client code.